Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp

```diff
@@ -4,11 +4,113 @@
 #include "Widgets/PSOverlayWidget.h"
 
 #include "Components/Image.h"
+#include "Curves/CurveFloat.h"
+#include "Data/PSDataAsset.h"
 
 #include UE_INLINE_GENERATED_CPP_BY_NAME(PSOverlayWidget)
 
-// Sets the visibility of the background overlay and lock icon.
-void UPSOverlayWidget::SetOverlayVisibility(ESlateVisibility VisibilitySlate)
+// Sets the visibility of the overlay elements and playing fade animation if needed
+void UPSOverlayWidget::SetOverlayVisibility(ESlateVisibility VisibilitySlate, bool bShouldPlayFadeAnimation/* = false*/)
+{
+	if (!bShouldPlayFadeAnimation)
+	{
+		SetOverlayItemsVisibility(VisibilitySlate);
+		return;
+	}
+
+	bShouldPlayFadeAnimationInternal = bShouldPlayFadeAnimation;
+
+	if (VisibilitySlate == ESlateVisibility::Visible)
+	{
+		ESlateVisibility PrevOverlayOpacity = PSCBackgroundOverlay->GetVisibility();
```

Comment:
PSCBackgroundOverlay might crash

Reply:
added ensure to check the overlay items https://github.com/h4rdmol/ProgressionSystem/pull/20/commits/5491bf32d4235f92a009aa3024fd8c3f734ceeb3

Reply:
if (!ensureMsgf(pscBackgroundOverlay, TEXT("ASSERT: [%i] %s:\n'pscBackgroundOverlay' is not valid!"), __LINE__, *FString(__FUNCTION__))
    || !ensureMsgf(pscBackgroundIconLock, TEXT("ASSERT: [%i] %s:\n'pscBackgroundIconLock' is not valid!"), __LINE__, *FString(__FUNCTION__)))
{
    return;
}